File: ResourceManager/Models/Configurations/RolePermissionConfiguration.cs
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;
using ResourceManager.Configuration;

namespace ResourceManager.Models.Configurations
{
    public class RolePermissionConfiguration : IEntityTypeConfiguration<RolePermission>
    {
        public void Configure(EntityTypeBuilder<RolePermission> builder)
        {
            builder.ToTable("RolePermissions");
            builder.HasKey(x => new { x.RoleID, x.PermissionID });

            builder.HasOne(x => x.Permission)
                .WithMany(x => x.RolePermissions)
                .HasForeignKey(x => x.PermissionID);

            builder.HasOne(x => x.Role)
                .WithMany(x => x.RolePermissions)
                .HasForeignKey(x => x.RoleID);

            builder.HasData(DesignData.Instance.RolePermissions);
        }
    }
}

File: ResourceManager/Models/Configurations/RolePermissionConfiguration.vb
Imports Microsoft.EntityFrameworkCore.Metadata.Builders
Imports Microsoft.EntityFrameworkCore
Imports ResourceManager.Configuration

Namespace ResourceManager.Models.Configurations
    Public Class RolePermissionConfiguration
        Implements IEntityTypeConfiguration(Of RolePermission)
        Public Sub Configure(ByVal builder As EntityTypeBuilder(Of RolePermission)) Implements IEntityTypeConfiguration(Of RolePermission).Configure
            builder.ToTable("RolePermissions")
            builder.HasKey(Function(x) New With {x.RoleID, x.PermissionID
            })

            builder.HasOne(Function(x) x.Permission).WithMany(Function(x) x.RolePermissions).HasForeignKey(Function(x) x.PermissionID)

            builder.HasOne(Function(x) x.Role).WithMany(Function(x) x.RolePermissions).HasForeignKey(Function(x) x.RoleID)

            builder.HasData(DesignData.Instance.RolePermissions)
        End Sub
    End Class
End Namespace
